src/ui/Folding/Folding.ts

```diff
@@ -334,29 +352,45 @@ export class Folding extends Component {
       if (resultNodes[i].result.uniqueId == uniqueId) {
         return resultNodes[i];
       }
-      let resultNode = Folding.findUniqueId(resultNodes[i].children, uniqueId);
+      const resultNode = Folding.findUniqueId(resultNodes[i].children, uniqueId);
       if (resultNode != null) {
         return resultNode;
       }
     }
     return null;
   }
 
+  private switcherooFoldingFields() {
+    // Swap "old" childField and parentField and assign them to the "new" parent option
+    // This needs to be done because connectors push the default data in *reverse* order compared to what the index expect.
+    if (this.options.childField != null) {
+      this.logger.warn('Detecting usage of deprecated option "childField". Assigning it automatically to the "parent" option instead.');
```

Comment:
I feel like this message will be confusing to users because they will not understand why we need to do this. Maybe we need to add something to briefly explain what's going on. Something like 'the option definition changed blabla..'.

As a user if I saw this in the console I would question myself as to whether this is actually intended :P